lighthouse-core/report/v2/renderer/logger.js

```diff
@@ -0,0 +1,71 @@
+/**
```

Comment:
IDK. I'm just porting :)

Knowing the separation of LH proper and integrations of LH is still unclear. Right now the operating assumption is that we're keeping everything we currently have b/c https://github.com/GoogleChrome/lighthouse/issues/1937 doesn't capture these types of details.

For me, the dream is that there will be all sorts of existing tools that adopt a LH report. It's impossible to foresee all those integration touch points. Rather than designing our system around a particular _integration_ , it might be easier to flip it. IOW, the integration (e.g. devtools) decides what parts of LH it wants to throw away or replace. Removing handlebars, adding `templateContext` are a step towards that.

For the features stuff, I've updated it to take the approach in https://github.com/GoogleChrome/lighthouse/commit/b12f2ea70a4f786138c24730d6d569f77d26dbfc. Devtools can decide not to load the feature (including logger). 

---

Some open thoughts/questions for everyone. "**Integration X**" can be substituted for  "devtools".

- The core report should contain the features we want to show up in the most places. For example, if you run LH in the CLI/extension/WPT/Gulliver/LHaaS, you'll get the UI we vend (logger, export, etc.).
- What happens if you generate an html report in **Integration X** and open it later? Do you see LH's base UI or their version?
- What happens if you open that same report in Viewer?
- CSS is the easiest way to get rid of base UI: `#lh-log { display: none}`. We're already [doing that today](https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/report/styles/report.css#L63).
   - Alternatively, **Integration X** loads its own stylesheets and effectively changes the entire report.
- If you also want to nuke code, don't inject `ReportUIFeature` ([here](https://github.com/GoogleChrome/lighthouse/pull/2000/files#diff-8423c2d032b8588d5009d6660287d634R40)), or stub files: `class Logger { log() {} }`. We could even provide those files for people if it makes sense.